x-pack/legacy/plugins/siem/public/components/ml/conditional_links/remove_kql_variables.test.ts

```diff
@@ -4,9 +4,28 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { removeKqlVariables } from './remove_kql_variables';
+import { removeKqlVariables, removeKqlVariablesUsingRegex, replacer } from './remove_kql_variables';
 
 describe('remove_kql_variables', () => {
+  describe('replacer function', () => {
+    test("should return empty string when passed `undefined` for 'parts'", () => {
+      let fakeParts; // test with 'undefined'
+      const replacedResult = replacer(' and user.name: "$user.name$"', fakeParts);
```

Comment:
You can pass in optionally a `undefined` instead of using a `let` variable. Most people use this way of testing `undefined`.
```ts
const replacedResult = replacer(' and user.name: "$user.name$"', undefined);
```

Reply:
the type checker is still complaining about passing in undefined when `string` was expected

Reply:
This is fixed now as I gave the updated arguments to the function which were incorporated in this PR.